gateway-ha/pom.xml

```
@@ -613,6 +590,31 @@
                     </execution>
                 </executions>
             </plugin>
+
+            <!-- If two plugins are bound to the same phase, the order of execution is the same as the order in which you define them
+            Therefore, it is important to put "copy static" after "pnpm build", otherwise there is no dist/ folder to copy -->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-resources-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>copy static</id>
+                        <goals>
+                            <goal>copy-resources</goal>
+                        </goals>
+                        <phase>generate-resources</phase>
```

```suggestion
                        <phase>process-resources</phase>
```

That should fix it without relying on the order

---

This wouldn't work.

```
[2024-05-08T17:45:51.645Z] #11 76.27 [INFO] --- resources:3.3.1:resources (default-resources) @ gateway-ha ---
[2024-05-08T17:45:51.645Z] #11 76.28 Downloading from root-repos: https://artprod.dev.bloomberg.com/artifactory/root-repos/org/apache/maven/shared/maven-filtering/3.3.1/maven-filtering-3.3.1.pom
[2024-05-08T17:45:51.905Z] #11 76.31 Progress (1): 6.0 kB
                    
Downloaded from root-repos: https://artprod.dev.bloomberg.com/artifactory/root-repos/org/apache/maven/shared/maven-filtering/3.3.1/maven-filtering-3.3.1.pom (6.0 kB at 195 kB/s)
[2024-05-08T17:45:51.905Z] #11 76.31 Downloading from root-repos: https://artprod.dev.bloomberg.com/artifactory/root-repos/org/apache/maven/shared/maven-filtering/3.3.1/maven-filtering-3.3.1.jar
[2024-05-08T17:45:51.905Z] #11 76.33 Progress (1): 7.5/55 kB
Progress (1): 13/55 kB 
Progress (1): 30/55 kB
Progress (1): 46/55 kB
Progress (1): 55 kB   
                   
Downloaded from root-repos: https://artprod.dev.bloomberg.com/artifactory/root-repos/org/apache/maven/shared/maven-filtering/3.3.1/maven-filtering-3.3.1.jar (55 kB at 2.0 MB/s)
[2024-05-08T17:45:51.905Z] #11 76.36 [INFO] Copying 3 resources from src/main/resources to target/classes
[2024-05-08T17:45:51.905Z] #11 76.36 [INFO] 
[2024-05-08T17:45:51.905Z] #11 76.36 [INFO] --- resources:3.3.1:copy-resources (copy static) @ gateway-ha ---
[2024-05-08T17:45:51.905Z] #11 76.37 [INFO] Copying 4 resources from ../webapp/dist to src/main/resources/static
[2024-05-08T17:45:51.905Z] #11 76.37 [INFO] 
```
`(default-resources)` is called before `copy-resources` is called, resulting `src/main/resources/static` not being included in `target/classes`, therefore it's not built in jar

---

We can have another execution to force copy `src/main/resources/static` to `target/classes` tho, it might work. But I'm not sure if it's gonna be cleaner

---

Maybe we can shuffle the pnpm call and usage into an even earlier phase